src/openms/CMakeLists.txt

```diff
@@ -86,7 +86,7 @@ include (${PROJECT_SOURCE_DIR}/includes.cmake)
 
 #------------------------------------------------------------------------------
 # all the dependency libraries are linked into libOpenMS.so
-set(OPENMS_DEP_LIBRARIES Evergreen LibSVM::LibSVM XercesC::XercesC Qt5::Core Qt5::Network)
+set(OPENMS_DEP_LIBRARIES Evergreen LibSVM::LibSVM XercesC::XercesC Eigen3::Eigen Qt5::Core Qt5::Network)
```

Comment:
@jpfeuffer if we want to leverage Eigen functionalities then it needs to be a normal DEP library I think.

Reply:
Only if you want to expose its types in the public API

Reply:
it is header only and if we want to use its functionality then we need to provide access to the matrix class. Like I did with the getEigenMatrix . Right?

Reply:
But public dependencies are dependencies used by us AND downstream applications/libraries.

Downstream (=our own tools or external tools/libs) will need the headers only if some of the types are in our public interface.

Reply:
I think they are, e.g. we also exposed the matrix through pyOpenMS. One might be able to do that that but I think this would imply that we hide the Eigen Matrix type and need to provide an extensive interface.

Reply:
Yes but before, the data was passed around in our matrix class (as vecs of vecs).
And only internally converted to Eigen.
Now, with the public convertToEigen, and the Eigen Matrix member, it will indeed be in the public API.

Not saying that it's necessarily bad but you should be aware of that implication. You can pimpl it to avoid that.